pkg/security/security_profile/activity_tree/activity_tree.go

```diff
@@ -293,10 +293,18 @@ func (at *ActivityTree) insertEvent(event *model.Event, dryRun bool, generationT
 		return false, fmt.Errorf("invalid event: %s", err)
 	}
 
-	node, newProcessNode, err := at.CreateProcessNode(event.ProcessCacheEntry, generationType, dryRun, resolvers)
+	createProcessNodeDryRun := dryRun
+	if event.GetEventType() != model.ExecEventType {
+		createProcessNodeDryRun = true // do not insert a process node if we want to insert another event type
+	}
+	node, newProcessNode, err := at.CreateProcessNode(event.ProcessCacheEntry, generationType, createProcessNodeDryRun, resolvers)
 	if err != nil {
 		return false, err
 	}
+	if event.GetEventType() != model.ExecEventType && newProcessNode {
```

Comment:
Should we add a test on `createProcessNodeDryRun` ? Like `&& createProcessNodeDryRun == true` ?

Reply:
I don't think it's needed: if `event.GetEventType() != model.ExecEventType`, then `createProcessNodeDryRun` is necessarily `== true`

Reply:
Sorry I think what I meant is add `&& createProcessNodeDryRun == false` => it's OK to add a process node from a DNS event, if `createProcessNodeDryRun == true`. My point is that we don't always want to exit here.

Reply:
I just pushed a new version with the modification we agreed on! It should be fine now. Thanks for pointing me the issue.